homeassistant/components/cast/helpers.py

```diff
@@ -155,3 +167,139 @@ def invalidate(self):
         else:
             self._mz_mgr.deregister_listener(self._uuid, self)
         self._valid = False
+
+
+class PlaylistError(Exception):
+    """Exception wrapper for pls and m3u helpers."""
+
+
+class PlaylistSupported(PlaylistError):
+    """The playlist is supported by cast devices and should not be parsed."""
+
+
+@dataclass
+class PlaylistItem:
+    """Playlist item."""
+
+    length: str | None
+    title: str | None
+    url: str
+
+
+def _is_url(url):
+    """Validate the URL can be parsed and at least has scheme + netloc."""
+    result = urlparse(url)
+    return all([result.scheme, result.netloc])
+
+
+async def _fetch_playlist(hass, url):
+    """Fetch a playlist from the given url."""
+    try:
+        session = aiohttp_client.async_get_clientsession(hass, verify_ssl=False)
+        async with session.get(url, timeout=5) as resp:
+            playlist_data = (await resp.content.read(64 * 1024)).decode()
```

Comment:
`decode` only works if it's UTF-8 text. Would it make sense to also check/return the content type?

Reply:
We can be pretty sure that the content type is set correctly as other media players also fail on that

Reply:
```suggestion
            if resp.content_type not in ("…", "…"):
                raise PlaylistError(f"Unsupported content type: {resp.content_type}")
            playlist_data = (await resp.content.read(64 * 1024)).decode()
```

Reply:
Fixed in 1adddb5 by decoding with the charset indicated by the server and raising if that fails